prost-types/src/lib.rs

```diff
@@ -907,4 +917,38 @@ mod tests {
         // Must contain at least one "/" character.
         assert_eq!(TypeUrl::new("google.protobuf.Duration"), None);
     }
+
+    #[test]
+    fn check_timestamp_ord() {
```

Comment:
Maybe you could add a parameterized test. This way you could easily test many scenarios . Here is an example: https://github.com/tokio-rs/prost/blob/a1ab898a9bfed407cd005e6655529a6577fb0371/prost-types/src/lib.rs#L733-L805